### ISSUE TITLE
ETL/audit: detect malformed-MPO JPEGs (data quality) — cover original AND resized/ paths

### ISSUE THREAD

**Issue by yeelauren** [data-mermaid/mermaid-segmentation#165]
## Problem

During the DINOv3 baseline run (#129), training logs surface PIL warnings:

```
UserWarning: Image appears to be a malformed MPO file, it will be interpreted as a base JPEG file
```

These images still **decode** (PIL falls back to reading them as a base JPEG), so they pass the ETL/resize corrupted-image validation and are **not** flagged today — but a malformed MPO is a real data-quality signal we should detect and remediate at the source.

The training-time warning only tells us the issue *exists* (Python dedups it to ~once per DataLoader worker); it does not tell us **which** images are affected. To act on it we need a per-`(source_id, image_id)` inventory.

## Ask

Add malformed-MPO (and ideally general "image decodes with warnings / non-standard format") detection to the CoralNet ETL/audit, producing a flag in the audit output so affected images can be remediated per source.

## Must consider the `resized/` paths too

Detection cannot look only at the original images. Two reasons:

1. **Training reads the resized copies.** The baseline now reads via the training manifest's `image_s3_key`, which resolves to the resized copy (`<output_prefix>/resized/s{source_id}/images/{image_id}.jpg`, see `_resized_s3_key_for` in `mermaidseg/datasets/coralnet/preprocessing/resize.py`) for images above threshold, and the original otherwise. So the warnings observed at train time come from whichever path the manifest points at.
2. **The resize fast-path passes some images through unchanged.** `resize_image_to_threshold` returns the original bytes untouched when the image is already a small JPEG (`mermaidseg/datasets/coralnet/preprocessing/resize.py`), while larger images are re-encoded via `Image.save(..., "JPEG")`. So a malformed-MPO file **may or may not persist** into its resized copy depending on which path it took — the malformed-MPO status of the original and the resized copy can differ.

Therefore the audit should check the **actual keys consumed** (original + `resized/`), not assume the two are equivalent.

## Suggested scope

- [ ] Detect malformed-MPO / warns-on-decode images (e.g. inspect for the MPO/APP2 marker or capture the PIL warning during a decode probe).
- [ ] Run detection over **both** the original prefix (`<source_s3_prefix>/s{id}/images/{id}.jpg`) and the `resized/` prefix, and record per `(source_id, image_id)` which copies are affected.
- [ ] Emit a `malformed_mpo` (or broader `image_format_warning`) column in the audit parquet + a per-source count.
- [ ] Decide remediation: normalize on ingest (force a clean re-encode so downstream reads are warning-free) vs. flag-and-skip vs. leave-and-report.

## Notes

- Alternative quick signal (run-scoped, not a fix): capture offending `image_s3_key`s into the training data-load failure report via `BaseCoralDataset._record_load_failure` / `failure_report_path`. Useful for a one-off inventory, but the durable fix belongs in the ETL/audit.
- Surfaced from #129; the baseline intentionally leaves the PIL warning **un-suppressed** so this stays visible until addressed.
